programs/bpf_loader/src/syscalls.rs

```diff
@@ -1121,7 +1121,11 @@ impl<'a, 'b> SyscallObject<BpfError> for SyscallSha256<'a, 'b> {
                     ),
                     result
                 );
-                let cost = compute_budget.sha256_byte_cost * (val.len() as u64 / 2);
+                let cost = compute_budget.syscall_base_cost.max(
+                    compute_budget
+                        .sha256_byte_cost
+                        .saturating_mul(val.len() as u64 / 2),
```

Comment:
don't we need to prevent the multiplicand from being one or zero for it to have any affect?

also not sure why we aren't adding to the base, or is it in fact a floor?

Reply:
The cost is consumed right after calculation in both places in this function, so functionally an addition

Reply:
I do think we could and probably use a lower value than `syscall_base_cost` here

Reply:
> I do think we could and probably use a lower value than `syscall_base_cost` here

I changed it to mem_op_base_cost.  Ideally, the logic here would punish abusive behavior quickly and not overcharge legitimate hashing requests. I don't know what are the legitimate patterns of hashing slices.  Maybe if the loop computes hash for many 0 length slices, the cost should increase dramatically? For example after 1000 iterations bump the base cost to something like 10000? Would that impact impact legitimate hashing? Maybe a well-behaved program optimizes its hashing requests into a single syscall with many slices?

Reply:
Good ideas, feel free to improve the punishment :-)

Reply:
I added an exponential abuse factor based on the number of iterations over slices.